### PR TITLE
Observation forms content

### PR DESCRIPTION
**Pull request by Nayira**
# Functionality Added
Implemented EOBS-1549, EOBS-1550, EOBS-1551, EOBS-1553, EOBS-1560, EOBS-1561, EOBS-1562
Added new version of Node for Travis (from 0.10 to 6.11.1)
# Functionality Removed
N/A
# Possible Side Effects of Changes
N/A
# Checks
- [x] Linting passing
- [x] Documentation up to date
- [x] Timings on expected conditions reviewed
- [x] Deprecated selectors reviewed
- [x] Code re-usability reviewed

### CODE REVIEW

File: steps/mobile_common_actions.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+from liveobs_ui.page_object_models.mobile.patient_page import PatientPage
+from liveobs_ui.page_object_models.mobile.data_entry_page import DataEntryPage
+from liveobs_ui.selectors.mobile.get_selector_by_something import \
+    get_element_selector
+from liveobs_ui.page_object_models.mobile.mobile_common import BaseMobilePage
+from liveobs_ui.selectors.mobile.patient_page_selectors import \
+    ADHOC_OBS_MENU_BUTTON
+
+
+@given("they view the {page_select} list")
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: We should look to remove the need to have multiple decorators as we could potentially end up with 3 decorators on each function to match `given they view the patient list`, `when they view the patient list` and `then they view the patient list.

Does behave have like an `@any` decorator or something to simplify this?

File: steps/mobile_common_actions.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    """
+    Finds a button from it's name/parameter and clicks it
+
+    :param context: Behave context
+    :param button_name:  name of the button to be pressed
+    :return: clicks the button
+    """
+    button_find = BaseMobilePage(context.driver)
+    button = button_find.find_button_to_select(button_name)
+    button.click()
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Use `click_and_verify_change` in selenium lib for this. We've already for the page object model in use so should be simple refactor

File: steps/mobile_common_actions.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    :param observation_field: name of the observation field to be checked
+    """
+    field_check = DataEntryPage(context.driver)
+    assert field_check.verify_obs_field_displayed(observation_field), \
+        "Expected field '{}' to be displayed but it isn't.".format(
+            observation_field)
+
+
+@then('the {observation_field} field is not displayed')
+def verify_obs_field_is_not_displayed(context, observation_field):
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Feel like this and the above step could be merged into one matcher which switches if it's displayed or not based on the text on the step

File: steps/mobile_common_actions.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    :param context: Behave context
+    :param observation_field: name of the observation field to be checked
+    """
+    field_check = DataEntryPage(context.driver)
+    assert field_check.verify_obs_field_not_displayed(observation_field), \
+        "Expected field '{}' to not be displayed but it is.".format(
+            observation_field)
+
+
+@then('the {obs_data_entry_field} entry field is {state_set} to '
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Might be worth adding some notion of a type to these matches as to ensure that when we start looking at other things to do with the entry field (such as error state etc) we don't run into issues here.

Page on how to use multi-method step matcher to reuse steps with different inferred types 
https://jenisys.github.io/behave.example/step_matcher/use_multi_methods.html

File: steps/mobile_common_actions.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    """
+    Finds a specific input field and checks the correctness of its expected
+    attribute state
+
+    :param context: Behave context
+    :param obs_data_entry_field: Name of the field to be checked
+    :param state_set: specifies the true/false state to verify
+    :param expected_state_type: Either 'Mandatory' or 'Necessary'
+    """
+    field_selector = get_element_selector(
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: Didn't we add a method somewhere to return the element from a selector name? If we did then we should use this here to cut down the lines

File: steps/mobile_common_actions.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+@when('the value {value} is selected in the {input_field} field')
+def select_value_in_field(context, value, input_field):
+    """
+
+    :param context:
+    :param value:
+    :param input_field:
+    :return:
+    """
+    field_selector = get_element_selector(
```
Comment (Gimpneek) [inline steps/mobile_common_actions.py]: This step matcher could be merged with the above with a switch on if we do an action on a select or input as they do the same thing but the last line

File: requirements.txt [hunk 7]
```diff
@@ -2,4 +2,6 @@ flake8==3.3.0
 pylint==1.7.1
 selenium==3.4.2
 sphinx==1.6.1
-git+git://github.com/BJSS/BJSS_liveobs_selenium.git
\ No newline at end of file
+git+git://github.com/BJSS/BJSS_liveobs_selenium.git@master
+git+git://github.com/behave/behave.git@master
```
Comment (Nayira) [inline requirements.txt]: We'll fix this later - doesn't apply to this PR

File: steps/mobile_common_actions.py (hunk 2 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: Done with you

File: steps/mobile_common_actions.py (hunk 1 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: Agreed to ignore this for now - to fix later

File: steps/mobile_common_actions.py (hunk 3 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: Merged the 2 steps together.

File: steps/mobile_common_actions.py (hunk 4 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: Created tech debt to deal with this in the future (EOBS-1559)

File: steps/mobile_common_actions.py (hunk 5 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: Done it :shipit:

File: steps/mobile_common_actions.py (hunk 6 above)
Comment (Nayira) [inline steps/mobile_common_actions.py]: done

**Pull request merged**